src/save/write.ts

```diff
@@ -115,12 +132,17 @@ export function writeSync(
  */
 function getDataToWrite(
   destinationPath: string,
-  image: Image,
-  options?: WriteOptionsPng | WriteOptionsJpeg | WriteOptions,
+  image: Image | Mask,
+  options?: WriteOptionsBmp | WriteOptionsPng | WriteOptionsJpeg | WriteOptions,
 ): Uint8Array {
   if (!options || !('format' in options)) {
     const extension = nodePath.extname(destinationPath).slice(1).toLowerCase();
     if (extension === 'png' || extension === 'jpg' || extension === 'jpeg') {
+      if (image instanceof Mask) {
+        image = image.convertColor('GREY');
+      }
+      return encode(image, { ...options, format: extension });
+    } else if (extension === 'bmp' && image instanceof Mask) {
```

Comment:
Should not check for Mask here. Let the call to `encodeBmp` handle it (see my other comment)